src/main/java/hudson/matrix/MatrixChildParametersAction.java

```diff
@@ -61,6 +62,17 @@
         return parameters;
     }
 
+    public void onLoad(MatrixRun run){
```

Comment:
This deserves a comment on why are we doing this. We are replacing the most straightforward way with something somewhat surprising.